catalog/transformer/catalog-transformer-pdf/src/main/java/ddf/catalog/transformer/input/pdf/PdfInputTransformer.java

```diff
@@ -81,26 +63,39 @@
     private static final FastDateFormat DATE_FORMAT = FastDateFormat.getInstance(
             "yyyy-MM-dd'T'HH:mm:ssZZ");
 
-    private Map<ServiceReference, ContentMetadataExtractor> contentMetadataExtractors =
-            Collections.synchronizedMap(new TreeMap<>(new ServiceComparator()));
+    private CheckedFunction<InputStream, PDDocument> inputStreamPDDocumentFunction =
+            PDDocument::load;
 
-    public void addContentMetadataExtractors(
-            ServiceReference<ContentMetadataExtractor> contentMetadataExtractorRef) {
-        Bundle bundle = getBundle();
-        if (bundle != null) {
-            ContentMetadataExtractor cme = bundle.getBundleContext()
-                    .getService(contentMetadataExtractorRef);
-            contentMetadataExtractors.put(contentMetadataExtractorRef, cme);
-        }
+    private CheckedFunction<PDDocument, String> geoParserFunction = GEO_PDF_PARSER::getWktFromPDF;
+
+    private CheckedFunction<PDDocument, byte[]> thumbnailFunction = this::generatePdfThumbnail;
+
+    private MetacardType metacardType;
```

Comment:
Never mind, just saw all the setters. Looks like those are needed only for unit testing. Any chance we could replace those with injected objects and/or factories? If not, maybe have a package-private constructor and inject those there, that way we would be able to make those `final`?